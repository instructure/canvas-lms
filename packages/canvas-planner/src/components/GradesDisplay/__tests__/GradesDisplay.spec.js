/*
 * Copyright (C) 2018 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import {shallow} from 'enzyme'
import {GradesDisplay} from '../index'

it('renders some course grades', () => {
  const mockCourses = [
    {
      id: '1',
      shortName: 'Ticket to Ride 101',
      color: 'blue',
      href: '/courses/1',
      score: null,
      grade: null,
      hasGradingPeriods: true
    },
    {
      id: '2',
      shortName: 'Ingenious 101',
      color: 'green',
      href: '/courses/2',
      score: 42.34,
      grade: 'D',
      hasGradingPeriods: false
    },
    {
      id: '3',
      shortName: 'Settlers of Catan 201',
      color: 'red',
      href: '/courses/3',
      score: 'blahblah',
      grade: null,
      hasGradingPeriods: false
    }
  ]
  const wrapper = shallow(<GradesDisplay courses={mockCourses} />)
  expect(wrapper).toMatchSnapshot()
})

it('does not render caveat if no courses have grading periods', () => {
  const mockCourses = [
    {
      id: '1',
      shortName: 'Ticket to Ride 101',
      color: 'blue',
      href: '/courses/1',
      score: null,
      grade: null,
      hasGradingPeriods: false,
      enrollmentType: 'StudentEnrollment'
    }
  ]
  const wrapper = shallow(<GradesDisplay courses={mockCourses} />)
  expect(wrapper).toMatchSnapshot()
})

it('renders a loading spinner when loading', () => {
  const mockCourses = [
    {
      id: '1',
      shortName: 'Ticket to Ride 101',
      color: 'blue',
      href: '/courses/1',
      score: null,
      grade: null,
      hasGradingPeriods: true
    },
    {
      id: '2',
      shortName: 'Ingenious 101',
      color: 'green',
      href: '/courses/2',
      score: 42.34,
      grade: 'D',
      hasGradingPeriods: false
    }
  ]
  const wrapper = shallow(<GradesDisplay loading courses={mockCourses} />)
  expect(wrapper).toMatchSnapshot()
})

it('renders an ErrorAlert if there is an error loading grades', () => {
  const mockCourses = [
    {id: '1', shortName: 'Ticket to Ride 101', color: 'blue', href: '/courses/1'}
  ]
  const wrapper = shallow(<GradesDisplay courses={mockCourses} loadingError="There was an error" />)
  expect(wrapper).toMatchSnapshot()
})
