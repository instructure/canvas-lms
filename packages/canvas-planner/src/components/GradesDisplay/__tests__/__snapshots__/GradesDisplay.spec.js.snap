// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render caveat if no courses have grading periods 1`] = `
<View
  display="auto"
>
  <View
    display="auto"
    textAlign="center"
  >
    <Heading
      border="none"
      color="inherit"
      ellipsis={false}
      level="h2"
      margin="0 0 large 0"
    >
      <Text
        as="span"
        letterSpacing="normal"
        size="medium"
        weight="bold"
      >
        My Grades
      </Text>
    </Heading>
  </View>
  <View
    as="div"
    display="auto"
    margin="0 0 large 0"
  >
    <div
      style={
        Object {
          "borderBottom": "solid thin",
          "borderBottomColor": "blue",
        }
      }
    >
      <Link
        as="button"
        ellipsis={false}
        href="/courses/1"
        iconPlacement="start"
        linkRef={[Function]}
        variant="default"
      >
        <Text
          as="span"
          color="primary"
          letterSpacing="normal"
          size="small"
          transform="uppercase"
        >
          Ticket to Ride 101
        </Text>
      </Link>
    </div>
    <Text
      as="div"
      letterSpacing="normal"
      size="large"
      weight="light"
    >
      No Grade
    </Text>
  </View>
</View>
`;

exports[`renders a loading spinner when loading 1`] = `
<View
  display="auto"
>
  <View
    display="auto"
    textAlign="center"
  >
    <Heading
      border="none"
      color="inherit"
      ellipsis={false}
      level="h2"
      margin="0 0 large 0"
    >
      <Text
        as="span"
        letterSpacing="normal"
        size="medium"
        weight="bold"
      >
        My Grades
      </Text>
    </Heading>
  </View>
  <View
    as="div"
    display="auto"
    margin="0 0 large 0"
    textAlign="center"
  >
    <Spinner
      as="div"
      size="small"
      title="Grades are loading"
      variant="default"
    />
  </View>
</View>
`;

exports[`renders an ErrorAlert if there is an error loading grades 1`] = `
<View
  display="auto"
>
  <ErrorAlert
    error="There was an error"
  >
    Error loading grades
  </ErrorAlert>
  <View
    display="auto"
    textAlign="center"
  >
    <Heading
      border="none"
      color="inherit"
      ellipsis={false}
      level="h2"
      margin="0 0 large 0"
    >
      <Text
        as="span"
        letterSpacing="normal"
        size="medium"
        weight="bold"
      >
        My Grades
      </Text>
    </Heading>
  </View>
</View>
`;

exports[`renders some course grades 1`] = `
<View
  display="auto"
>
  <View
    display="auto"
    textAlign="center"
  >
    <Heading
      border="none"
      color="inherit"
      ellipsis={false}
      level="h2"
      margin="0 0 large 0"
    >
      <Text
        as="span"
        letterSpacing="normal"
        size="medium"
        weight="bold"
      >
        My Grades
      </Text>
    </Heading>
  </View>
  <View
    as="div"
    display="auto"
    margin="0 0 large 0"
  >
    <div
      style={
        Object {
          "borderBottom": "solid thin",
          "borderBottomColor": "blue",
        }
      }
    >
      <Link
        as="button"
        ellipsis={false}
        href="/courses/1"
        iconPlacement="start"
        linkRef={[Function]}
        variant="default"
      >
        <Text
          as="span"
          color="primary"
          letterSpacing="normal"
          size="small"
          transform="uppercase"
        >
          Ticket to Ride 101
        </Text>
      </Link>
    </div>
    <Text
      as="div"
      letterSpacing="normal"
      size="large"
      weight="light"
    >
      No Grade
    </Text>
  </View>
  <View
    as="div"
    display="auto"
    margin="0 0 large 0"
  >
    <div
      style={
        Object {
          "borderBottom": "solid thin",
          "borderBottomColor": "green",
        }
      }
    >
      <Link
        as="button"
        ellipsis={false}
        href="/courses/2"
        iconPlacement="start"
        linkRef={[Function]}
        variant="default"
      >
        <Text
          as="span"
          color="primary"
          letterSpacing="normal"
          size="small"
          transform="uppercase"
        >
          Ingenious 101
        </Text>
      </Link>
    </div>
    <Text
      as="div"
      letterSpacing="normal"
      size="large"
      weight="light"
    >
      42.34%
    </Text>
  </View>
  <View
    as="div"
    display="auto"
    margin="0 0 large 0"
  >
    <div
      style={
        Object {
          "borderBottom": "solid thin",
          "borderBottomColor": "red",
        }
      }
    >
      <Link
        as="button"
        ellipsis={false}
        href="/courses/3"
        iconPlacement="start"
        linkRef={[Function]}
        variant="default"
      >
        <Text
          as="span"
          color="primary"
          letterSpacing="normal"
          size="small"
          transform="uppercase"
        >
          Settlers of Catan 201
        </Text>
      </Link>
    </div>
    <Text
      as="div"
      letterSpacing="normal"
      size="large"
      weight="light"
    >
      No Grade
    </Text>
  </View>
  <View
    as="div"
    display="auto"
    textAlign="center"
  >
    <Text
      as="span"
      fontStyle="italic"
      letterSpacing="normal"
      size="x-small"
    >
      *Only most recent grading period shown.
    </Text>
  </View>
</View>
`;
